indexing-service/src/main/java/org/apache/druid/indexing/common/task/CompactionTask.java

```diff
@@ -452,6 +454,45 @@ public boolean isPerfectRollup()
     return tuningConfig != null && tuningConfig.isForceGuaranteedRollup();
   }
 
+  /**
+   * In MSQ, MVDs need conversion to array for rollup, and they aren't supported as partition columns. This check
+   * therefore is particularly for MSQCompactionRunner; it returns true when dimension types aren't known
+   * from CompactionTask spec, or if either rollup or partition dimensions contain any string-type column.
+   */
+  boolean needMultiValuedDimensions()
```

Comment:
Can this be renamed to something like `needToIdentifyMultiValuedDimensions` or `shouldAnalyzeMultiValuedDimensions` (to correspond to segment analyzer)?
Just `needMvd` seems ambiguous.

Reply:
Updated to `identifyMultiValuedDimensions`